sdk/monitor/monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,389 @@
+# Troubleshooting Azure Monitor Query client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure Monitor Query client library for JavaScript.
+
+## Table of contents
+
+- [General troubleshooting](#general-troubleshooting)
+  - [Enable client logging](#enable-client-logging)
+  - [Troubleshooting authentication issues with logs and metrics query requests](#authentication-errors)
+- [Troubleshooting logs query](#troubleshooting-logs-query)
+  - [Troubleshooting insufficient access error](#troubleshooting-insufficient-access-error-for-logs-query)
```

Comment:
I'm torn because I love that the links all make sense on their own (for tabbing between them directly / accessibility purposes), but the word `Troubleshooting` starts to melt my eyeballs by the end of the TOC. Could we maybe omit it for the sub-bullets?

Reply:
updated